### PR TITLE
Update deprecated Prometheus Metrics

### LINKED ISSUE #30584: K8s: Cilium Agent: Grafana Dashboards: L7 forwarded request panel shows no data
### Is there an existing issue for this?

- [X] I have searched the existing issues

I have deployed Cilium via its Helm chart in the `kube-system` namespace of a _kind_ cluster.
I have created L7 `CiliumNetworkPolicy` k8s resources.
I have enabled Prometheus metrics for the Cilium Agent component.

### `values.yaml`
```yaml
policyEnforcementMode: always
hubble:
  metrics:
    enabled:
    - dns:query;ignoreAAAA
    - drop
    - tcp
    - flow
    - icmp
    - http
  relay:
    enabled: true
    prometheus:
      enabled: true
  ui:
    enabled: true
prometheus:
  enabled: true
operator:
  prometheus:
    enabled: true
```
I have deployed the [Prometheus & Grafana example](https://github.com/cilium/cilium/tree/main/examples/kubernetes/addons/prometheus) provided in the repository. 
I want to view information on Grafana regarding the L7 CNP k8s resources in my cluster.
I click on _Search dashboards_ in the left pane, and go the _Cilium Metrics_ dashboard under the _General_ folder.
I turn to the **L7 forwarded request** panel.

### What I expected to happen
Graphical data should be appearing on the panel, showing information relating to L7 forwarded/denied requests.

### What happened
The panel is empty, showing the `No data` message.

### Investigation
By editing the panel, the Prometheus metrics it draws data from appear to be the following:

```
cilium_policy_l7_denied_total
cilium_policy_l7_forwarded_total
cilium_policy_l7_received_total
```

Indeed, the [Grafana dashboard definition](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/files/cilium-agent/dashboards/cilium-dashboard.json#L4694) utilizes these Prometheus metrics to populate the L7 forwarded requests panel with data relating to the CNP/CCNP k8s resources created inside the cluster.

However, based on the [associated documentation](https://docs.cilium.io/en/stable/observability/metrics/#policy-l7-http-kafka-fqdn), only the `cilium_policy_l7_total` Prometheus metric is currently supported.

### Evidence
Indeed, after looking at the [Upgrade Guide](https://github.com/cilium/cilium/blob/main/Documentation/operations/upgrade.rst#removed-metrics-1), it becomes evident that the above mentioned metrics along with `cilium_policy_l7_parse_errors_total` have been removed.

### Recommendation
1. Reconstruct the panel to use the `sum(rate(cilium_policy_l7_total{k8s_app="cilium", pod=~"$pod"}[1m]))` PromQL query, while removing the now redundant queries.
2. Update the `cilium-dashboard.json` file referenced above to use the currently-supported PromQL query.
3. Update the [example](https://github.com/cilium/cilium/blob/main/examples/kubernetes/addons/prometheus/monitoring-example.yaml#L4926) Grafana dashboard definition. 

Alternatively, If providing this panel is no longer relevant for Cilium Agent, perhaps consider removing it from the _Cilium Metrics_ Grafana dashboard.

The `cilium_policy_l7_parse_errors_total` metric is also used in the **Proxy response time (Avg)** and **Proxy response time (Max)** panels. Consider updating or repurposing them as well, so as to not include deprecated queries that produce void results.

### Cilium Version

```
cilium-cli: v0.15.20 compiled with go1.21.6 on darwin/arm64
cilium image (default): v1.14.5
cilium image (stable): v1.14.6
cilium image (running): 1.14.6
```

### Kernel Version

`Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:18 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6000 arm64`

### Kubernetes Version

```
Client Version: v1.29.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.27.3
```

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by karojohn**
Fixes: #30584 

Update PromQL queries that use obsolete metrics, belonging to panels of the _Cilium Metrics_ Grafana dashboard.

### CODE REVIEW

**Review by sayboras — APPROVED** [review]

**Comment by karojohn** [comment]
@joamaki please review at your earliest convenience.

**Review by joamaki — APPROVED** [review]

**Comment by karojohn** [comment]
@sayboras looks like some workflows require maintainer approval. Should they first be completed successfully before merging?

**Pull request merged**
